source/http_connection.c

```diff
@@ -108,22 +144,25 @@ napi_value aws_napi_http_connection_new(napi_env env, napi_callback_info info) {
         return NULL;
     }
 
-    if (napi_get_value_external(env, node_args[0], (void **)&options.bootstrap)) {
+    struct aws_nodejs_client_bootstrap *node_bootstrap = NULL;
+    if (napi_get_value_external(env, node_args[0], (void **)&node_bootstrap)) {
         napi_throw_error(env, NULL, "Unable to extract bootstrap from external");
         return NULL;
     }
 
     struct aws_napi_callback on_connection_setup;
     AWS_ZERO_STRUCT(on_connection_setup);
-    if (!aws_napi_is_null_or_undefined(env, node_args[1])) {
-        if (aws_napi_callback_init(
-                &on_connection_setup,
-                env,
-                node_args[1],
-                "aws_http_connection_on_connection_setup",
-                s_http_on_connection_setup_params)) {
-            return NULL;
-        }
+    if (aws_napi_is_null_or_undefined(env, node_args[1])) {
+        napi_throw_error(env, NULL, "2nd argument (on_connection_setup) must be a callback");
+        return NULL;
+    }
+    if (aws_napi_callback_init(
+            &on_connection_setup,
+            env,
+            node_args[1],
```

Comment:
We don't clean up the result of THIS `aws_napi_callback_init()` call if the NEXT call fails a few lines down

Reply:
It doesn't need to be cleaned up. A failed init results in a clean structure.